api/saml.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"encoding/xml"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/crewjam/saml"
+	"github.com/crewjam/saml/samlsp"
+)
+
+// getSAMLServiceProvider generates a new service provider object with the
+// (optionally) provided descriptor (metadata) for the identity provider.
+func (a *API) getSAMLServiceProvider(identityProvider *saml.EntityDescriptor) *saml.ServiceProvider {
+	externalURL, err := url.ParseRequestURI(a.config.API.ExternalURL)
+	if err != nil {
+		panic(err)
+	}
+
+	externalURL.Path = strings.TrimSuffix(externalURL.Path, "/")
+	externalURL.Path += "/auth/v1/"
```

Comment:
should this be made into a config or handled outside of gotrue? this seems specific to supabase?

Reply:
Yeah, good catch!